mlrun/feature_store/feature_set.py

```diff
@@ -559,6 +556,95 @@ def upsert_feature(name):
+        """add feature aggregation rule
+
+        example::
+
+            myset.add_aggregation(column="ask", operations=["sum", "max"], windows="1h", period="10m")
+
+        :param column:     name of column/field aggregate. If several aggregations are added for the same column,
+                            either they must have the same windows/period or a "name" must be provided
+        :param operations: aggregation operations, e.g. ['sum', 'std']
+        :param windows:    time windows, can be a single window, e.g. '1h', '1d',
+                            or a list of same unit windows e.g ['1h', '6h']
+                            windows are transformed to fixed windows or
+                            sliding windows depending whether period parameter
+                            provided.
+
+                            - Sliding window is fixed-size overlapping windows
+                              that slides with time.
+                              The window size determines the size of the sliding window
+                              and the period determines the step size to slide.
+                              Period must be integral divisor of the window size.
+                              If the period is not provided then fixed windows is used.
+
+                            - Fixed window is fixed-size, non-overlapping, gap-less window.
+                              The window is referred to as a tumbling window.
+                              In this case, each record on an in-application stream belongs
+                              to a specific window. It is processed only once
+                              (when the query processes the window to which the record belongs).
+
+        :param period:     optional, sliding window granularity, e.g. '10m'
+        :param name:       optional, aggregation name/prefix. Must be unique per feature set.If not passed,
+                           name will be column
+        :param step_name: optional, graph step name
+        :param state_name: *Deprecated* - use step_name instead
+        :param after:      optional, after which graph step it runs
+        :param before:     optional, comes before graph step
+        """
+        return self._add_aggregation(
+            column=column,
+            operations=operations,
+            windows=windows,
+            name=name,
+            period=period,
+            step_name=step_name,
+            after=after,
+            before=before,
+            state_name=state_name,
+        )
+
+    def add_aggregation(
+        self,
+        name,
+        column,
+        operations,
+        windows,
+        period=None,
+        step_name=None,
+        after=None,
+        before=None,
+        state_name=None,
+    ):
+        warnings.warn(
+            "This method is deprecated. Use add_feature_aggregation instead (name parameter is optional)",
+            # TODO: In 0.9.0 do changes in examples & demos In 0.11 remove
```

Comment:
You already did the changes in the examples & demos (which might be problematic, cause a user looking on the docs know will try to use `add_feature_aggregation` by they won't have this function in any released version), I don't know the exact status of that function and feature store in general:
1. If you feel like it's still somewhat beta - you can probably already just remove the old `add_aggregation` function
2. If you feel like backwards compatible is needed (probably the answer) - I suggest reverting the docs changes, scheduling (here in the TODO) to do them in 0.10.0, and remove in 0.11.0

Reply:
@yaronha @dinal your opinion on the above is welcome

Reply:
@theSaarco

Reply:
Since we already have users working with feature-store and `add_aggregation` is pretty commonly used, then backwards compatibility is definitely needed. So, doing (1) is not the way to go.
As for the docs - since these will probably get updated before the release is out (which seems to be an issue anyway), I agree this is an issue and probably option (2) is needed - maybe separate those changes to a different PR and not merge it until the time is right.

Reply:
i don't see ANY reason to change the name, please keep `add_aggregation`, its clear and its short and its already used/understood.

Reply:
@yaronha we can't keep the name cause the function signature has changed. "name" is now optional so it has to move to the end of the args list after the mandatory args

Reply:
@Hedingber we need to keep backwards compatibility for sure cause we don't want to break this very used api.
the docs update will always be an issue and we can't keep track on when is the right timing to merge these changes (right before GA?) 
users should look at the exact docs version they are using. like https://docs.mlrun.org/en/v0.8.0/, and not latest, or at least be aware that their might be some changes for the newer features

Reply:
@Hedingber we need to keep backwards compatibility for sure cause we don't want to break this very used api.
the docs update will always be an issue and we can't keep track on when is the right timing to merge these changes (right before GA?) 
users should look at the exact docs version they are using. like https://docs.mlrun.org/en/v0.8.0/, and not latest, or at least be aware that their might be some changes for the newer features

Reply:
@dinal this is no reason to change such a fundamental API off course we add `feature aggregation`  on a feature set, another way to address it is to accept `None` for all attrs and raise if they are not provided. we already have lots of demos and snips of code out there, a rather not change, especially when the name doesnt make sense